docs/api.md

```diff
@@ -53,13 +53,79 @@ In our experience, any system that is successful needs to grow and change as new
+
+Not all kubernetes clusters will support all API groups.  However, we intend that if an API server
+supports a group, it supports all the Kinds in the group, and all of them at the same version(s).
+
 ## API versioning
 
-To make it easier to eliminate fields or restructure resource representations, Kubernetes supports multiple API versions, each at a different API path prefix, such as `/api/v1beta3`. These are simply different interfaces to read and/or modify the same underlying resources. In general, all API resources are accessible via all API versions, though there may be some cases in the future where that is not true.
+To make it easier to eliminate fields or restructure resource representations, Kubernetes supports multiple API versions, each at a different API path prefix, such as `/api/v1beta3`.
 
 We chose to version at the API level rather than at the resource or field level to ensure that the API presents a clear, consistent view of system resources and behavior, and to enable controlling access to end-of-lifed and/or experimental APIs.
 
-The [API and release versioning proposal](design/versioning.md) describes the current thinking on the API version evolution process.
+Note that API versioning and Software versioning are only indirectly related.  The [API and release
+versioning proposal](design/versioning.md) describes the current thinking on the API version
+evolution process.
+
+Different API versions imply different levels of support and stability:
+ - Alpha level:
+   - The version names contain `alpha` (e.g. `v1alpha1`).
+   - The object schema may change significantly in the next API version.
+   - We may not give any tools or instructions for migrating your objects from one version to the next
+     &emdash; you may have to delete, rewrite, and re-create your objects when you upgrade your cluster to the next
+     version.
+   - Support for object/feature may be dropped entirely in next software release.
+   - Do not count on overlapping support for "previous" and "current" version; supported version
+     may change abruptly on a new software release.
+   - Flag changes may be needed to enable alpha versions.
+   - Hosted clusters (e.g. GKE) may not support alpha versions.
+   - Code that implements alpha objects may be less well tested.  Enabling alpha support may active bugs
+     even if you do not create alpha versioned objects.
+   - Please do try alpha versions in test clusters and give feedback, as significant changes are
+     possible during the Alpha level.
+ - Beta level:
+   - The version names contain `beta` (e.g. `v2beta3`).
+   - The object schema is not expected to change significantly in the next API version, but occasionally it
+     might.
+   - We will give advice and/or tools for migrating your objects from one beta version to the next
+     beta version and to the stable version.
+   - We expect overlapping support for "previous" and "current" version for several software releases,
```

Comment:
I'm not sure we expect it for several releases (since releases are 3 months apart).